src/webview/components/options-widget.tsx

```diff
@@ -117,11 +130,48 @@ export class OptionsWidget extends React.Component<OptionsWidgetProps, {}> {
         return errors;
     };
 
+    componentDidUpdate(prevProps: Readonly<OptionsWidgetProps>, prevState: Readonly<OptionsWidgetState>): void {
+        if (this.props.initialTitle !== prevProps.initialTitle) {
+            this.setState({ title: this.props.initialTitle });
+        }
+        if (!prevState.isTitleEditing && this.state.isTitleEditing) {
+            this.labelEditInput.current?.focus();
+            this.labelEditInput.current?.select();
+        }
+    }
+
     override render(): React.ReactNode {
         this.formConfig.initialValues = this.optionsFormValues;
+        const isLabelEditing = this.state.isTitleEditing;
 
         return (
             <div className='memory-options-widget px-4'>
+                <div className='title-container'>
+                    <InputText
+                        ref={this.labelEditInput}
+                        type='text'
+                        value={this.state.title}
+                        onChange={this.handleTitleEdit}
```

Comment:
It might be possible to simplify the state handling here (and avoid storing `previousTitle`) by not storing the state of the input as `title` and handling its changes. Instead, set the `defaultValue` to the current title, let the user do what they want, and then only enact the state change in the `title` field in the `confirmEditedTitle` method using the `ref` for the text input.

Then `title` would always be passed in as props, and the only job of this widget would be calling the `updateTitle` method when we're sure the user wants a new title.

Reply:
Excellent suggestion! Done! :+1: